src/components/edit/thematic/ThematicDialog.js

```diff
@@ -213,6 +223,21 @@ export class ThematicDialog extends Component {
                 .forEach(ou => loadOrgUnitPath(ou.id));
         }
 
+        // Set the default classification/legend for selected data item without visiting the style tab
+        if (columns !== prev.columns) {
+            const dataItem = getDataItemFromColumns(columns);
+
+            if (dataItem) {
```

Comment:
in which case will this be false?

Reply:
It will be false when item type, or item group is changed.